### PR TITLE
FW-2740 [Important] Add Form inputs in chat transcript 

### PR DESCRIPTION
**Pull request by Ojashwa**
### What do you want to achieve?
- Add support of Form inputs in chat transcript.

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [ ] I have compared it with mocks and all design elements are the same.
- [x] I have tested it in IE Browser.

### How was the code tested?

-  Use this new property `"postFormDataAsMessage": "true"` inside the `action ` of form templete.
```
"action": {
      "message": "Submit details",
      "requestType": "postBackToBotPlatform",
      "postFormDataAsMessage": "true",
      "formAction": "https://reqres.in/api/users"
  }
```
- Click on send transcript to user button in the conversation section and check for form data in the chat transcript.
---

https://user-images.githubusercontent.com/22856546/132954722-b7d1ebdc-182f-4351-bbbf-06cc5213000b.mov

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 1]
```diff
@@ -671,7 +681,12 @@ Kommunicate.richMsgEventHandler = {
             (messagePxy['metadata'] = msgMetadata);
         (Object.keys(msgMetadata).length > 0 ||
             Object.keys(messagePxy).length > 0) &&
-            Kommunicate.sendMessage(messagePxy);
+        Kommunicate.sendMessage(messagePxy);
+        postFormDataAsMessage && 
+        Kommunicate.sendMessage({
```
Comment (devashishmamgain) [inline webplugin/js/app/km-rich-text-event-handler.js:686]: There is a send message above it, are we sending 2 messages for one form submission?

File: webplugin/js/app/kommunicate.js [hunk 2]
```diff
@@ -521,7 +521,6 @@ $applozic.extend(true, Kommunicate, {
             return 'km-fixed-container';
         }
     },
-    processPaymentRequest: function (options) {},
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate.js:524]: Where was it used?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Review by devashishmamgain — CHANGES_REQUESTED** [review]
Only 1 message is required, combine both if configured

File: webplugin/js/app/kommunicate.js (hunk 2 above)
Comment (Ojashwa) [reply webplugin/js/app/kommunicate.js:524]: nowhere, only declaration was found.

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 3]
```diff
@@ -556,6 +556,9 @@ Kommunicate.richMsgEventHandler = {
         var postBackToKommunicate = isActionableForm
             ? JSON.parse(target.dataset.postBackToKommunicate.toLowerCase())
             : false;
+        var ispostBackAsMessage = isActionableForm
```
Comment (devashishmamgain) [inline webplugin/js/app/km-rich-text-event-handler.js:559]: remove is, postBackAsMessage as is fine

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 4]
```diff
@@ -654,8 +657,19 @@ Kommunicate.richMsgEventHandler = {
         }
         var messagePxy = {};
         var msgMetadata = {};
-        replyText && (messagePxy.message = replyText); //message to send
-
+        var postFormDataAsMessage =
+            ispostBackAsMessage &&
+            data &&
```
Comment (devashishmamgain) [inline webplugin/js/app/km-rich-text-event-handler.js:662]: can you share a sample for how the data looks like? is it an array?
couldn't understand why data need to be converted to JSON first and then remove the JSON attributes from it?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-rich-text-event-handler.js (hunk 4 above)
Comment (Ojashwa) [reply webplugin/js/app/km-rich-text-event-handler.js:662]: image

Comment (devashishmamgain) [reply webplugin/js/app/km-rich-text-event-handler.js:662]: directly convert object to the format you need, json stringify is not required

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Review by ChannuBS — APPROVED** [review]

**Pull request merged**